### PR TITLE
html-css-practice initial commit

### PR DESCRIPTION
**Pull request by linkqwd**
#### html-css-practice

[Demo](https://linkqwd.github.io/Google-pop-up-menu__pure-css/)

[Styles.css](https://github.com/linkqwd/Google-pop-up-menu__pure-css/blob/master/src/css/styles.css)

One thing I struggled with - is ` :focus ` on pop-up-menu elements.
It can be simply implemented with something like:
```
input:focus + label-selector {
    outline: 1px solid #666;;
}
```
or using ` :focus-within ` on parent element;
But in both cases, those pop-menu-items will triggered focus state not only by tabbing and pressing space (which is usual behavior), but also by mouse-clicking! Which looks strange.

I currently don't know and wasn't able to find solution to handle this problem in pure css and html.

So for now checkboxe's (menu-pop-up's) can be reachable by tabbing but not focused with outline or something similar.

### CODE REVIEW

File: submissions/linkqwd/html-css-practice/src/index.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	<title>Google pop-up menu</title>
+	<meta name="description" content="">
+	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
+	<link rel="stylesheet" href="css/styles.css">
+</head>
+
+<body class="b-layout">
+	<ul class="b-menu">
+		<li class="b-menu__item">
+			<a class="b-menu__item-link" href="#">Gmail</a></li>
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/index.html]: Unnecessary repeat class hierarchy in the class name:
`.b-menu__item` -> `.b-menu__item-link`
better use simple/shorter name spacing: `.b-menu__link`

File: submissions/linkqwd/html-css-practice/src/index.html [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		<li class="b-menu__item">
+			<a class="b-menu__item-link" href="#">Images</a></li>
+		<li class="b-menu__item b-menu__item_type_apps">
+			<div class="b-toggle">
+				<input type="checkbox" id="toggle-apps" class="b-toggle__input-drop-down">
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/index.html]: Same problem with name spacing too long.
`.toggle__drop-down` will be better.

File: submissions/linkqwd/html-css-practice/src/index.html [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				<label for="toggle-apps" class="b-toggle__label"></label>
+				<div class="b-toggle__content-wrapper b-menu__drop-down">
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/index.html]: I'll see you use BEM. This block is `element` of block `toggle` and in the same time it's `element` of block `menu`.
Chose, it must be element of existing block or independent block inside existing element?

File: submissions/linkqwd/html-css-practice/src/index.html [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+					<ul class="b-apps">
+						<li class="b-apps__item">
+							<a class="b-apps__item-link b-apps__item-link_icon_docs" href="#">Docs</a>
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/index.html]: For cases like this better use modificators `.apps__link--docs`.

File: submissions/linkqwd/html-css-practice/src/index.html [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+						</li>
+						<li class="b-apps__item">
+							<a class="b-apps__item-link b-apps__item-link_icon_bussiness" href="#">My Bussiness</a>
+						</li>
+						<input type="checkbox" id="toggle-more" class="b-toggle__input-more">
+						<label for="toggle-more" class="b-toggle__show-more">More</label>
+						<li class="b-apps__item b-toggle__hidden">
+							<a class="b-apps__item-link b-apps__item-link_icon_blogger" href="#">Blogger</a>
+						</li>
+						<li class="b-apps__item b-toggle__hidden">
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/index.html]: Element state better put inside modificators `.apps__item--hidden`

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.b-apps {
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps.scss]: Too much for this class. Better put this property into another file with abstract general abstract properties.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps__item-link_icon_account.scss [hunk 7]
```diff
@@ -0,0 +1,5 @@
+.b-apps__item-link_icon_account {
+    &:before {
+        background: url(../img/v1_051523630.png) -4px 2799px;
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps__item-link_icon_account.scss]: Adding several background-images better using custom mixin.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps__item-link_icon_account.scss [hunk 8]
```diff
@@ -0,0 +1,5 @@
+.b-apps__item-link_icon_account {
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps__item-link_icon_account.scss]: If you want separate adding background images better do it in one file. Add one image in one file it's too much.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps__item-link.scss [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    min-height: 75px;
+    line-height: 1.2;
+    border: 1px solid transparent;
+    box-sizing: border-box;
+
+    &:hover {
+        border-color: #ddd;
+    }
+
+    &:before {
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/b-apps/b-apps__item-link.scss]: Pceudo-element need to write with double colon: `::after`.
Singe colon in pceudo-element it's old syntax.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/b-toggle__input-drop-down.scss [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    opacity: 0;
+    position: absolute;
+    top: -10000px;
+
+    &:checked ~ .b-toggle__content-wrapper, &:checked ~ .b-menu__item-decoration-triangle {
+        opacity: 1;
+        visibility: visible;
+    }
+
+    // alternative solution for focusing
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/b-toggle__input-drop-down.scss]: Do not commit commented code.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/index.scss [hunk 11]
```diff
@@ -0,0 +1,11 @@
+@import 'b-toggle';
+@import 'b-toggle__input-drop-down';
+@import 'b-toggle__label';
+@import 'b-toggle__content-wrapper';
+@import 'b-toggle__show-more';
+@import 'b-toggle__input-more';
+@import 'b-toggle__hidden';
+@import 'b-toggle__hidden-block-wrap';
+
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/index.scss]: Remove empty lines.

**Comment by linkqwd** [comment]
Thanks for review

File: submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/index.scss (hunk 11 above)
Comment (linkqwd) [reply submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/index.scss]: Sorry, here I don't understand. 
In first comment you said that I'm supposed to add a new line at the end of every file, and here is opposite.

Comment (A-Ostrovnyy) [reply submissions/linkqwd/html-css-practice/src/sass/blocks/b-toggle/index.scss]: In this case, the current file have 2 empty lines at the end. 
End of every file must be like the end of the README.md

File: submissions/linkqwd/html-css-practice/src/sass/blocks/apps/apps__item_state_hidden.scss [hunk 12]
```diff
@@ -0,0 +1,3 @@
+.apps__item_state_hidden {
+    display: none;
+}
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/apps/apps__item_state_hidden.scss]: Add an empty line at the end of the file.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/layout/index.scss [hunk 13]
```diff
@@ -0,0 +1 @@
+@import 'layout';
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/layout/index.scss]: Why you imported only one file inside another. Mabey better to import this file directly?

**Comment by A-Ostrovnyy** [comment]
Works well!
But Menu icon, bell icon, and avatar without outline/border on focus state. Also, button more need outline too. Underlined text on More button not enough to identify a focused element. People with problemed vision needs a more obvious sign. 
Just add outline and it will be done.

File: submissions/linkqwd/html-css-practice/src/sass/blocks/toggle/toggle__drop-down.scss [hunk 14]
```diff
@@ -7,4 +7,8 @@
         opacity: 1;
         visibility: visible;
     }
-}
+
+    &:focus + .toggle__label {
+        box-shadow: 0px 0px 0px 2px #a5c7fe;
+    }
+}
```
Comment (A-Ostrovnyy) [inline submissions/linkqwd/html-css-practice/src/sass/blocks/toggle/toggle__drop-down.scss]: Add an empty line at the end of the file.

**Comment by A-Ostrovnyy** [comment]
Nice work!

**Pull request merged**